### PR TITLE
Test dispute

### LINKED ISSUE #193: Tests for dispute

### PR DESCRIPTION
**Pull request by Yashk767**
# Description

Added tests for dispute function

Fixes #193 

# Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [ ] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

File: cmd/dispute.go [hunk 1]
```diff
@@ -13,37 +13,37 @@ import (
 
 var giveSortedAssetIds []int
 
-func HandleDispute(client *ethclient.Client, config types.Configurations, account types.Account, epoch uint32, razorUtils utilsInterface, proposeUtils proposeUtilsInterface) {
-	numberOfProposedBlocks, err := utils.GetNumberOfProposedBlocks(client, account.Address, epoch)
+func HandleDispute(client *ethclient.Client, config types.Configurations, account types.Account, epoch uint32, razorUtils utilsInterface, proposeUtils proposeUtilsInterface, cmdUtils utilsCmdInterface, blockManagerUtils blockManagerInterface, transactionUtils transactionInterface) error {
```
Comment (ashish10677) [inline cmd/dispute.go:16]: Seems like too many parameters, can we consolidate all the utils into one struct and then pass it here?
Would be a better approach.

File: cmd/dispute.go [hunk 2]
```diff
@@ -58,19 +58,20 @@ func HandleDispute(client *ethclient.Client, config types.Configurations, accoun
 		}
 	}
 	giveSortedAssetIds = []int{}
+	return nil
 }
 
-func Dispute(client *ethclient.Client, config types.Configurations, account types.Account, epoch uint32, blockId uint8, assetId int) error {
-	blockManager := utils.GetBlockManager(client)
-	numOfStakers, err := utils.GetNumberOfStakers(client, account.Address)
+func Dispute(client *ethclient.Client, config types.Configurations, account types.Account, epoch uint32, blockId uint8, assetId int, razorUtils utilsInterface, cmdUtils utilsCmdInterface, blockManagerUtils blockManagerInterface, transactionUtils transactionInterface) error {
```
Comment (ashish10677) [inline cmd/dispute.go:64]: Same thing here as well.

File: cmd/dispute_test.go [hunk 3]
```diff
@@ -0,0 +1,370 @@
...
+				if err.Error() != tt.want.Error() {
+					t.Errorf("Error for Dispute function, got = %v, want = %v", err, tt.want)
+				}
+			}
+
+		})
+	}
+}
+
+//func TestGiveSorted(t *testing.T) {
```
Comment (ashish10677) [inline cmd/dispute_test.go:177]: Remove this if not required.

**Review by ashish10677 — CHANGES_REQUESTED** [review]

File: cmd/dispute.go (hunk 1 above)
Comment (Yashk767) [reply cmd/dispute.go:16]: Yes we can do this.

**Review by ashish10677 — APPROVED** [review]

**Pull request merged**